packages/wallet/src/redux/reducers/index.ts

```diff
@@ -1,133 +1,56 @@
   }
   if (action.type === DISPLAY_MESSAGE_SENT) {
-    state = { ...state, displayOutbox: undefined };
+    nextOutbox.displayOutbox = undefined;
   }
-
   if (action.type === TRANSACTION_SENT_TO_METAMASK) {
-    state = { ...state, transactionOutbox: undefined };
-  }
-
-  const conclusionStateFromOwnRequest = receivedValidOwnConclusionRequest(state, action);
-  if (conclusionStateFromOwnRequest) {
-    return conclusionStateFromOwnRequest;
-  }
-
-  const conclusionStateFromOpponentRequest = receivedValidOpponentConclusionRequest(state, action);
-  if (conclusionStateFromOpponentRequest) {
-    return conclusionStateFromOpponentRequest;
+    nextOutbox.transactionOutbox = undefined;
   }
+  state = sideEffectsReducer(state, nextOutbox);
 
   switch (state.stage) {
     case INITIALIZING:
       return initializingReducer(state, action);
-    case OPENING:
-      return openingReducer(state, action);
-    case FUNDING:
-      return fundingReducer(state, action);
-    case RUNNING:
-      return runningReducer(state, action);
-    case CHALLENGING:
-      return challengingReducer(state, action);
-    case RESPONDING:
-      return respondingReducer(state, action);
-    case WITHDRAWING:
-      return withdrawingReducer(state, action);
-    case CLOSING:
-      return closingReducer(state, action);
+    case WALLET_INITIALIZED:
+      return initializedReducer(state, action);
     default:
       return unreachable(state);
   }
 };
 
-const receivedValidOwnConclusionRequest = (
-  state: WalletState,
-  action: WalletAction,
-): ApproveConclude | null => {
-  if (state.stage !== FUNDING && state.stage !== RUNNING) {
-    return null;
-  }
-  if (action.type !== CONCLUDE_REQUESTED || !ourTurn(state)) {
-    return null;
-  }
-  return approveConclude({ ...state, displayOutbox: showWallet() });
-};
-
-const receivedValidOpponentConclusionRequest = (
-  state: WalletState,
-  action: WalletAction,
-): AcknowledgeConclude | null => {
-  if (state.stage !== FUNDING && state.stage !== RUNNING) {
-    return null;
-  }
-  if (action.type !== COMMITMENT_RECEIVED) {
-    return null;
-  }
+export function sideEffectsReducer<T extends SharedWalletState>(
+  state: T,
+  sideEffects: OutboxState | undefined,
+): T {
+  // TODO: What about unhandled actions?
 
-  const { commitment, signature } = action;
-
-  if (commitment.commitmentType !== CommitmentType.Conclude) {
-    return null;
-  }
-  // check signature
-  const opponentAddress = state.participants[1 - state.ourIndex];
-  if (!validCommitmentSignature(commitment, signature, opponentAddress)) {
-    return null;
-  }
-  if (!validTransition(state, commitment)) {
-    return null;
+  if (!sideEffects) {
+    return state;
   }
+  // TODO: We need to think about whether to overwrite existing outbox items.
+  const newState = { ...state, outboxState: { ...state.outboxState } };
+  Object.keys(sideEffects).map(k => (newState.outboxState[k] = sideEffects[k]));
```

Comment:
A bit confused on what is intended here. `sideEffects` object is changed, but then the object is not used for the rest of the function.